archive/zstd.cpp

```diff
@@ -77,16 +78,19 @@ tl::expected<std::vector<std::uint8_t>, ZstdError> zstd_decode(std::span<uint8_t
 
         last_ret = ret;
         last_pos = out_buf.pos;
-        count++;
     }
 
-    assert(last_pos > 0);
-
     if (last_ret != 0) {
         return tl::unexpected{ZstdError::DecodeEarlyTermination};
     }
 
-    auto const out_size = (chunk_size * count) - (chunk_size * count - last_pos);
+    std::size_t out_size = 0;
+
+    if (count == 1) {
+        out_size = last_pos;
+    } else {
+        out_size = last_pos == 0 ? chunk_size * count : (chunk_size * count) - (chunk_size * count - last_pos);
```

Comment:
The falsy case in the ternary looks off; you sure it's correct?

`(chunk_size * count) - (chunk_size * count - last_pos)` can be simplified to `last_pos`.

Reply:
no, it's wrong. too much copy+paste from testing. fixing it now

Reply:
Also gonna add a test for multiples of chunk_size > 1, which would have caught this earlier

Reply:
Sounds good! I missed it the first review too. Good test cases is hard. :(

Reply:
Test case for `count != 1 && last_pos > 0` where this bug was?

Reply:
done